### PR TITLE
Nvfuser code bump 2_1_2022

### PR DESCRIPTION
**Pull request by jjsjann123**
Things changed in this PR that requires review:
1. aten/src/ATen/core/interned_strings.h
2. torch/csrc/jit/ir/alias_analysis.h : exposing createValue to allow efficient mutation
3. torch/csrc/jit/runtime/symbolic_shape_registry.cpp : added gelu/tanh/erf in registry
4. torch/jit/_script.py : throws scripting model sees autocast as decorator since it's not supported  

nvfuser code update:
1. codegen improvements and performance tuning
2. integration bug fixes for shape expression logic
3. kernel segmentation update to address perf regression from horizontal fusion
4. scalar cpu tensor promotion to support inter-device operation between cpu scalar tensor and cuda tensor

Things reverted from local changes:
aten::gelu with approximation (tracked in PR: https://github.com/pytorch/pytorch/pull/61439)

### CODE REVIEW

**Comment by jjsjann123** [comment]
Saw some cudnn failures which I believe is not relevant and should have already been fixed in upstream/master.

There's a core dump from here [linux-bionic-cuda10.2-py3.9-gcc7 / test (slow, 1, 1, linux.4xlarge.nvidia.gpu)](https://github.com/pytorch/pytorch/runs/5045496022?check_suite_focus=true): Don't know if it's related...
Log seems to suggest it's running te, so likely not `pytest benchmarks/fastrnns/test_bench.py --benchmark-sort=Name --benchmark-json=${BENCHMARK_DATA}/fastrnns_profiling_te.json --fuser=te --executor=profiling`

There are still tests queued but it seems to be stuck there for a while now.

File: torch/csrc/jit/ir/alias_analysis.h [hunk 1]
```diff
@@ -156,7 +156,7 @@ class AliasDb {
   // Copy `from`s aliasing info to `to`.
   void copyValue(Value* from, Value* to);
   // Create a new `value` that does not alias anything else.
-  void createValue(const Value* value);
+  TORCH_API void createValue(const Value* value);
```
Comment (eellison) [inline torch/csrc/jit/ir/alias_analysis.h]: should we add `TORCH_API` on the other two methods here as well ?

File: torch/csrc/jit/runtime/symbolic_shape_registry.cpp [hunk 2]
```diff
@@ -66,6 +66,9 @@ static const OperatorMap<std::string>& get_schema_to_function_graph() {
       {"aten::rsub.Tensor(Tensor self, Scalar other, Scalar alpha=1) -> Tensor", "unary"},
       {"aten::dropout(Tensor input, float p, bool train) -> Tensor", "unary"},
       {"aten::adaptive_avg_pool2d(Tensor self, int[2] output_size) -> Tensor", "adaptive_avg_pool2d"},
+      {"aten::gelu(Tensor self) -> Tensor", "unary"},
```
Comment (eellison) [inline torch/csrc/jit/runtime/symbolic_shape_registry.cpp]: These three ops are now in shape analysis via https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/symbolic_shape_registry_util.cpp#L39 so we don't need them here

**Review by eellison — APPROVED** [review]
JIT changes lgtm w/small nit

**Comment by jjsjann123** [comment]
Wondering if there's any issue on the import that we could be of help.

**Comment by jbschlosser** [comment]
@jjsjann123 It's in the process of landing, periodically hitting some spurious failures and re-attempting. Will let you know if I hit any real issues

**Comment by jbschlosser** [comment]
@jjsjann123 The tanh gelu PR #61439 finally landed - are there any changes needed here to account for this?

**Comment by jjsjann123** [comment]
Let me handle the conflicts. Thanks for landing the other PR.

**Comment by jjsjann123** [comment]
Tests passed locally. Should be good for another import.
Looks like there's some upstream failures introduced by other PR. Do you want me to follow that rebase as well?

**Comment by jbschlosser** [comment]
> Tests passed locally. Should be good for another import. Looks like there's some upstream failures introduced by other PR. Do you want me to follow that rebase as well?

Upstream failures are related to XLA, and the XLA-side PR https://github.com/pytorch/xla/pull/3039 is merged now. I think we're good to go; re-importing now.

**Pull request closed without merging**